src/Storage/Repository/InstanceRepository.cs

```diff
@@ -587,9 +587,33 @@ private async Task PostProcess(Instance instance)
         /// <param name="instances">the list of instances</param>
         private async Task PostProcess(List<Instance> instances)
         {
-            foreach (Instance item in instances)
+            Dictionary<string, Instance> instanceMap = instances.ToDictionary(key => key.Id, instance => instance);
+            var instanceGuids = instances.Select(i => i.Id).ToList();
+
+            foreach (Instance instance in instances)
+            {
+                instance.Id = $"{instance.InstanceOwner.PartyId}/{instance.Id}";
+            }
+
+            var dataElements = await _dataRepository.ReadAllForMultiple(instanceGuids);
+
+            foreach (var instanceGuid in instanceGuids)
+            {
+                instanceMap[instanceGuid].Data = dataElements[instanceGuid];
+            }
+
+            foreach (Instance instance in instances)
             {
-                await PostProcess(item);
+                instance.Id = InstanceIdToCosmosId(instance.Id);                
```

Comment:
InstanceId should be changed FROM cosmosId to regularId.
Check logic on line 570.

Reply:
It was the other way around of course.